gateway/handler.go

```diff
@@ -352,65 +351,78 @@ func (i *handler) getOrHeadHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	trace.SpanFromContext(r.Context()).SetAttributes(attribute.String("ResponseFormat", responseFormat))
 
-	resolvedPath, contentPath, ok := i.handlePathResolution(w, r, responseFormat, contentPath, logger)
-	if !ok {
+	// TODO: Why did the previous code do path resolution, was that a bug?
+	// TODO: Does If-None-Match apply here?
```

Comment:
I'm removing this comment. And yes, we did not need to perform path resolution for `application/vnd.ipfs.ipns-record`. That is something I missed when implementing that format.

Reply:
> Why did the previous code do path resolution, was that a bug?

I think I only implemented it that way in order for the `If-None-Match` header to be respected/set. But indeed we do not need to perform path resolution to provide IPNS record keys.

If we want to handle the `If-None-Match` here, we will probably have to implement some different logic for this case since the key is not based in the CID.

> (...) to be based on the hash of IPNS Record protobuf, and not the CID with libp2p-key.

Done, I've applied that suggestion.